be/src/exec/aggregator.cpp

```diff
@@ -804,37 +804,53 @@ Status Aggregator::_evaluate_const_columns(int i) {
 
 Status Aggregator::convert_to_chunk_no_groupby(ChunkPtr* chunk) {
     SCOPED_TIMER(_agg_stat->get_results_timer);
-    // TODO(kks): we should approve memory allocate here
-    auto use_intermediate = _use_intermediate_as_output();
-    Columns agg_result_column = _create_agg_result_columns(1, use_intermediate);
-    if (!use_intermediate) {
+    if (_agg_functions[0]->get_name() == "group_concat2") {
+        Columns agg_result_column = _create_agg_result_columns(_state->chunk_size(), false);
```

Comment:
Our function doesn't know the number of result rows in advance and the number is not a direct function of chunk_size(). Would this line be okay even in that case? It looks like the argument used in reserve() so it seems only for the performance.

Reply:
Yes, reserve is only for the performance here.